pkg/vm/vm.go

```diff
@@ -1804,6 +1804,28 @@ func throwUnhandledException(item stackitem.Item) {
 	panic(msg)
 }
 
+// ContractHasTryBlock checks if the currently executing contract has a TRY
+// block in one of its contexts.
+func (v *VM) ContractHasTryBlock() bool {
+	var topctx *Context // Currently executing context.
+	for i := 0; i < len(v.istack); i++ {
+		ictx := v.istack[len(v.istack)-1-i] // It's a stack, going backwards like handleException().
```

Comment:
maybe it'd be easier to invert `i` then

Reply:
Copy-pasting from `handleException` is more reliable!